spice/lib/vm_actions_rv.py

```diff
@@ -350,23 +350,24 @@ def rv_chk_con(vmi):
         hostname = socket.gethostname()    # See rv_url() function
         remote_ip = socket.gethostbyname(hostname)
     elif cfg.spice_proxy:
-        proxy_port = "3128"
-        if "http" in cfg.spice_proxy:
-            split = cfg.spice_proxy.split('//')[1].split(':')
+        prx_parse = cfg.spice_proxy.split('//')[-1]
+        if ']:' in prx_parse or '.' in prx_parse and ':' in prx_parse:
+            remote_ip, _, proxy_port = prx_parse.rpartition(':')
         else:
-            split = cfg.spice_proxy.split(':')
-        remote_ip = split[0]
-        if len(split) > 1:
-            proxy_port = split[1]
-        logger.info("Proxy port to inspect: %s", proxy_port)
+            remote_ip, proxy_port = prx_parse, "3128"
+        remote_ip = remote_ip.strip('[]')
+        if remote_ip.split('.')[-1].isalpha():
```

Comment:
I have spent good deal of time on this and verified on few test cases.  https://github.com/spiceqa/tp-spice/pull/56#discussion-diff-101510036R359 is for testing whether addresss in config is fqdn and I found he checking of TLD subscript as the easier check.